packages/poller/src/rebalance/callbacks.ts

```diff
@@ -165,10 +154,20 @@ export const executeDestinationCallbacks = async (context: ProcessingContext): P
         });
 
         // Update operation as completed with destination tx hash
-        const currentTxHashes = (operation.txHashes as TxHashes) || {};
+        const currentTxHashes = operation.transactions;
+        const destinationTx = currentTxHashes?.[route.destination] as
+          | TransactionEntry<{ receipt: TransactionReceipt }>
+          | undefined;
+        if (!destinationTx || !destinationTx.metadata || !destinationTx.metadata.receipt) {
+          logger.info('Origin transaction receipt not found for operation', { ...logContext, operation });
+          continue;
+        }
         await db.updateRebalanceOperation(operation.id, {
           status: RebalanceOperationStatus.COMPLETED,
-          txHashes: { ...currentTxHashes, destinationTxHash: tx.hash },
+          txHashes: {
```

Comment:
why not just `[route.destination]: tx.receipt`, with the `tx` from the `submitTransactionWithLogging` call above?